src/main/java/frc/robot/commands/sequences/BaseAutonSequence.java

```diff
@@ -38,24 +38,33 @@ public BaseAutonSequence(
     /**
      * Goes to a position and intakes a game piece.
      * @param intialPose The initial pose of the robot.
+     * @param waypoints The waypoints to hit along the path.
      * @param finalPose The destination pose of the robot.
      * @return The `SequentialCommandGroup` representing running the commands in order.
      */
-    protected Command goAndGrab(Pose2d initialPose, PiecePosition finalPose) {
-        return FollowPathCommand.from(swerveSubsystem, initialPose, List.of(), finalPose.getPose())
+    protected Command goAndGrab(Pose2d initialPose, List<Pose2d> waypoints, Pose2d finalPose) {
+        return FollowPathCommand.composedFrom(swerveSubsystem, initialPose, waypoints, finalPose)
             .alongWith(new TiltedElevatorCommand(tiltedElevatorSubsystem, ElevatorState.GROUND))
-            .alongWith(new RollerIntakeCommand(rollerSubsystem));
+            .andThen(new LockSwerveCommand(swerveSubsystem))
+            .andThen(new RollerIntakeCommand(rollerSubsystem).withTimeout(3));
```

Comment:
it seems unlikely that the drivetrain will place us in a perfect positoin where running the intake will grab a piece. i'm almost certain we'll need to run the intake while driving into the piece to pick one up